staging/src/k8s.io/apiserver/pkg/features/kube_features.go

```diff
@@ -188,23 +193,41 @@ func init() {
 // To add a new feature, define a key for it above and add it here. The features will be
 // available throughout Kubernetes binaries.
 var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
-	AdvancedAuditing:                    {Default: true, PreRelease: featuregate.GA},
-	APIResponseCompression:              {Default: true, PreRelease: featuregate.Beta},
-	APIListChunking:                     {Default: true, PreRelease: featuregate.Beta},
-	DryRun:                              {Default: true, PreRelease: featuregate.GA},
-	RemainingItemCount:                  {Default: true, PreRelease: featuregate.Beta},
-	ServerSideApply:                     {Default: true, PreRelease: featuregate.GA},
-	StorageVersionHash:                  {Default: true, PreRelease: featuregate.Beta},
-	StorageVersionAPI:                   {Default: false, PreRelease: featuregate.Alpha},
-	WatchBookmark:                       {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
-	APIPriorityAndFairness:              {Default: true, PreRelease: featuregate.Beta},
-	RemoveSelfLink:                      {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
-	SelectorIndex:                       {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
-	EfficientWatchResumption:            {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
-	APIServerIdentity:                   {Default: false, PreRelease: featuregate.Alpha},
-	APIServerTracing:                    {Default: false, PreRelease: featuregate.Alpha},
-	OpenAPIEnums:                        {Default: true, PreRelease: featuregate.Beta},
+	APIListChunking: {Default: true, PreRelease: featuregate.Beta},
```

Comment:
I think we could get rid of the need to list the features here with something like this:

```go
type FeatureSet struct{
  features map[Feature]FeatureSpec
}

func (f *FeatureSet) Register(feature Feature, default bool, prerelease prerelease) Feature {
  f.features[feature] = FeatureSpec{Default: default, PreRelease: prerelease}
  return feature
}
```

Then the feature declarations just call `Register` on the default feature set, e.g.
```diff
	// owner: @smarterclayton
	// alpha: v1.8
	// beta: v1.9
	//
	// Allow API clients to retrieve resource lists in chunks rather than
	// all at once.
-	APIListChunking featuregate.Feature = "APIListChunking"
+	APIListChunking = defaultKubernetesFeatureGates.Register("APIListChunking", true, featuregate.Beta)
```

I also like that this keeps the state next to the definition.

Reply:
I had the same idea, but didn't pursue it because of one drawback: `APIListChunking` has to become a global variable instead of a constant, right?

Is that relevant or can we trust all code to never write to these variables?

The `Register` also needs to be a bit more complicated or we need additional flavors of it because some feature gates also have `LockToDefault: true`.

Reply:
@tallclair: ping ^^^

Reply:
Maybe we should err on the side of separating sorting the file from refactoring the feature gate mechanism into separate PRs. Hard to review both at once.

Reply:
Yeah, I'm in favor of moving forward with this change and punting on the larger refactor.

@pohly that's a good point about non-const (I really with go had immutability...). You could probably do something hacky like:

```go
// ...
const APIListChunking featuregate.Feature = "APIListChunking"
var _ = defaultKubernetesFeatureGates.Register(APIListChunking, true, featuregate.Beta)
```

Note that `var _ =` is required to call the function outside a method, hence the hackiness.

Reply:
That's doable and looks like an improvement over the current approach. I can follow up with that in a second PR once this one has been merged.

Who can provide LGTM and approval?

Reply:
@liggitt ?? ^